### PR TITLE
get hashes, extract IOCs from strings, get filetype, global vt_api va…

### PR DESCRIPTION
**Pull request by deadbits**
Adds functions to:
- Get MD5, SHA1, SHA256 hash for file
- Leverage regex's to extract IPv4, IPv6, Email , and URL IOCs from a given string (instead of just providing the regex to user or offering verification functions, user can simply use `utilitybelt.extract_ipv4(big_string)`
- Verify that a file is valid (exists, is not a directory, size greater than zero, user has permissions to access it). This isn't DFIR specific but just a handy check if user will be passing files to other functions in utilitybelt.
- Get file type

### CODE REVIEW

File: utilitybelt/utilitybelt.py [hunk 1]
```diff
@@ -66,6 +66,80 @@
...
+def extract_ipv6(data):
+    """ Extract all ipv6 addresses in chunk of data """
+    ipv6 = re.findall(re_ipv6, data)
+    if ipv6 is not None and len(ipv6) > 0:
+        return ipv6
+
+
+def extract_emails(data):
+    """ Extract all email addresses in chunk of data """
+    emails = re.findall(re_email, data):
```
Comment (krmaxwell) [inline utilitybelt/utilitybelt.py]: Looks like there's a spurious trailing `:`

**Comment by sroberts** [comment]
TESTS FAILLLLLL!

**Comment by deadbits** [comment]
le sigh.

**Comment by deadbits** [comment]
I'm not sure what this error means in the tests. Any idea whats going on?

```
Submitting coverage to coveralls.io...
Source file is not python /home/ubuntu/utilitybelt/utilitybelt/utilitybelt.py
/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/requests/packages/urllib3/util/ssl_.py:100: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Coverage submitted!
Couldn't find a repository matching this job.

'url' coveralls returned exit code 2
```

**Comment by deadbits** [comment]
If anyone cares to figure out why this valid code is failing tests, I'd love to know so this can be finally merged

**Comment by sroberts** [comment]
Need to fix tests.

**Pull request closed without merging**